ndsl/quantity/quantity.py

```diff
@@ -347,12 +359,20 @@ def extent(self) -> tuple[int, ...]:
     @property
     def field_as_xarray(self) -> xr.DataArray:
         """Returns an Xarray.DataArray of the field (domain)"""
-        return xr.DataArray(self.field, dims=self.dims, attrs=self.attrs)
+        if isinstance(self.field, np.ndarray):
+            field = self.field
+        else:
+            field = self.field.get()
```

Comment:
Do you know how this works under the hood? It is definetly the right thing to do but I'm curious if transporting bounded array views from device to host is slow, as fast as copying the full thing or even faster? Because if slicing it was really slow, we might consider a `data.get()` followed by a cpu-slicing. But that is only if we see this being an issue

the xarray stuff is not on the critical path, so we can go either way, but I'm just curious.